public/app/features/explore/Explore.tsx

```diff
@@ -192,7 +193,7 @@ export class Explore extends React.PureComponent<ExploreProps> {
   refreshExplore = () => {
     const { exploreId, update } = this.props;
 
-    if (update.queries || update.ui || update.range || update.datasource) {
+    if (update.queries || update.ui || update.range || update.datasource || update.mode) {
```

Comment:
Changed so that we use value from state before url because of strange behaviour when using the Metrics/Logs switch.